### PR TITLE
Multiquery

### PR DESCRIPTION
**Pull request by uhbrar**
This implements the multiquery endpoint as discussed on the 4/1 ARAGORN call. The endpoint will take in a dict of AsyncQuery and then return a dict of the responses for the queries. All AsyncQuery passed in must use the same callback url. Additionally, the dict will utilize the same keys as the multiquery that is passed in. 

Example 1: 
- query: [multiquery1.txt](https://github.com/ranking-agent/strider/files/8410451/multiquery1.txt)
- response: [multiquery1 response.txt](https://github.com/ranking-agent/strider/files/8410455/multiquery1.response.txt)

Example 2: 
- query: [multiquery2.txt](https://github.com/ranking-agent/strider/files/8410489/multiquery2.txt)
- response: [multiquery2 response.txt](https://github.com/ranking-agent/strider/files/8410500/multiquery2.response.txt)

Utilizing this endpoint will allow all calls to use the same cache. There is a significantly smaller increase in processing time when using this endpoint as opposed to running each query individually.

### CODE REVIEW

**Comment by cbizon** [comment]
Just to confirm what I think this is implementing: 

1. A single call from aragorn to strider containing a dict of queries, and 
2. a single callback from strider to aragorn containing a dict of responses.

In the input, I think it's fine.  I guess I'm not sure why a dict vs a list though.

On the response, I think I'd rather either 
1. have strider do the merging and return a single response, or 
2. have strider return individual responses as they are generated, which could be fully trapi

Of the 2, I am coming around to the latter, which would let aragorn work on merging while strider is generating results.

**Comment by uhbrar** [comment]
You're right in both counts, yes. I went with a dict instead of a list just because I thought it would be easier to comprehend and look at, and it would be easier to reference the original query graph provided that way. Plus, the order doesn't really matter, so a dict seemed more appropriate. 

That first option is actually relatively easy to implement in strider, though, so I'll make that change and we can figure out which one is better.

**Comment by uhbrar** [comment]
One thing I'd like to mention. When strider returns, the whole process is done asynchronously, so the order in which responses are generated can not be guaranteed. For instance, if two queries are sent, the callback might receive either of them first, regardless of the order in the input.

**Comment by uhbrar** [comment]
Yeah, that's why I went with a dict for the input, since the order doesn't have meaning, I just felt like my initial explanation was unclear. 

This does present a bit of an issue though. How would ARAGORN differentiate between the responses, in that case? Would it just be comparing the query graphs?

**Comment by cbizon** [comment]
I guess it's not clear to me that ARAGORN needs to distinguish between the answers.  Part of the uncertainty here is that we are not sure what the return needs to look like for 'creative' responses. i.e. will we be returning the induced query graphs?  I don't think that's a given.

File: strider/server.py [hunk 1]
```diff
@@ -496,3 +547,36 @@ async def score_results(
...
+    """Handles multiple queries. Queries are sent back to a callback url as a dict with keys corresponding to keys of original query."""
+    query_keys = list(multiquery.keys())
+    callback = multiquery[query_keys[0]].dict().get("callback")
+    if not callback:
+        raise HTTPException(400, "callback url must be specified")
+    queries = {}
+    for query in query_keys:
+        query_dict = multiquery[query].dict()
+        query_callback = query_dict["callback"]
+        workflow = query_dict.get("workflow", None) or [{"id": "lookup"}]
```
[the remark this answers is not in the document]
Comment (uhbrar) [reply strider/server.py:567]: This becomes a bit more difficult with how we have to return things, so it isn't a good idea.

**Review by kennethmorton — APPROVED** [review]

**Pull request merged**